eth/eventdatahandler/handlers.go

```diff
@@ -0,0 +1,561 @@
+	logger := edh.logger.With(
+		fields.OperatorID(event.OperatorId),
+		zap.String("operator_pub_key", ethcommon.Bytes2Hex(event.PublicKey)),
+		zap.String("owner_address", event.Owner.String()),
+		zap.String("event_type", OperatorAdded),
+	)
+
+	logger.Debug("processing event")
+
+	od := &registrystorage.OperatorData{
+		PublicKey:    event.PublicKey,
+		OwnerAddress: event.Owner,
+		ID:           event.OperatorId,
+	}
+
+	// throw an error if there is an existing operator with the same public key and different operator id
+	if edh.operatorData.ID != 0 && bytes.Equal(edh.operatorData.PublicKey, event.PublicKey) && edh.operatorData.ID != event.OperatorId {
+		logger.Warn("malformed event: operator registered with the same operator public key",
+			zap.Uint64("expected_operator_id", edh.operatorData.ID))
+		return &MalformedEventError{Err: ErrAlreadyRegistered}
+	}
+
+	// TODO: consider saving other operators as well
+	exists, err := edh.nodeStorage.SaveOperatorData(txn, od)
+	if err != nil {
+		return fmt.Errorf("save operator data: %w", err)
+	}
+	if exists {
+		logger.Debug("operator data already exists")
+		return nil
+	}
+
+	if bytes.Equal(event.PublicKey, edh.operatorData.PublicKey) {
+		edh.operatorData = od
+	}
+
+	edh.metrics.OperatorPublicKey(od.ID, od.PublicKey)
+	logger.Debug("report operator public key",
+		fields.OperatorID(od.ID),
+		fields.PubKey(od.PublicKey))
+
+	logger.Debug("processed event")
+	return nil
+}
+
+func (edh *EventDataHandler) handleOperatorRemoved(txn basedb.Txn, event *contract.ContractOperatorRemoved) error {
+	logger := edh.logger.With(
+		fields.OperatorID(event.OperatorId),
+		zap.String("event_type", OperatorRemoved),
+	)
+	logger.Debug("processing event")
+
+	od, found, err := edh.nodeStorage.GetOperatorData(txn, event.OperatorId)
+	if err != nil {
+		return fmt.Errorf("could not get operator data: %w", err)
+	}
+	if !found || od == nil {
+		logger.Warn("malformed event: could not find operator data")
+		return &MalformedEventError{Err: ErrOperatorDataNotFound}
+	}
+
+	logger = logger.With(
+		zap.String("operator_pub_key", ethcommon.Bytes2Hex(od.PublicKey)),
+		zap.String("owner_address", od.OwnerAddress.String()),
+	)
+
+	// TODO: In original handler we didn't delete operator data, so this behavior was preserved. However we likely need to.
+	// TODO: Delete operator from all the shares.
+	//	var shares []Share
+	//	for _, s := range nodeStorage.Shares().List() {
+	//		// if operator in committee, delete him from it:
+	//		//     shares = aopend(shares, s)
+	//	}
+	//	nodeStorage.Shares().Save(shares)
+
+	logger.Debug("processed event")
+	return nil
+}
+
+func (edh *EventDataHandler) handleValidatorAdded(txn basedb.Txn, event *contract.ContractValidatorAdded) (*ssvtypes.SSVShare, error) {
+	logger := edh.logger.With(
+		zap.String("owner_address", event.Owner.String()),
+		zap.Uint64s("operator_ids", event.OperatorIds),
+		zap.String("operator_pub_key", ethcommon.Bytes2Hex(event.PublicKey)),
```

Comment:
@nkryuchkov , @pavelkrolevets this log is wrong. should be validator pubkey or pubkey. please go over tihs file and make sure logging is correct.

Reply:
@y0sher this is a typo in the log - `event.PublicKey` is a public key of the validator. Fixed.